include/nbl/builtin/hlsl/prefix_sum_blur/box_sampler.hlsl

```diff
@@ -0,0 +1,183 @@
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/enums.hlsl"
+#include "nbl/builtin/hlsl/macros.h"
+
+#ifndef _NBL_BUILTIN_BOX_SAMPLER_INCLUDED_
+#define _NBL_BUILTIN_BOX_SAMPLER_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace prefix_sum_blur
+{
+
+template<typename PrefixSumAccessor, typename T>
+struct BoxSampler
+{
+    using prefix_sum_accessor_t = PrefixSumAccessor;
+
+    PrefixSumAccessor prefixSumAccessor;
+    uint16_t wrapMode;
+    uint16_t linearSize;
+    T normalizationFactor;
+
+    T operator()(float32_t ix, float32_t radius, float32_t borderColor)
+    {
+        const float32_t alpha = frac(radius);
+        const float32_t rightIdx = float32_t(ix) + radius;
+        const float32_t leftIdx = float32_t(ix) - radius - 1;
+        const int32_t lastIdx = linearSize - 1;
+        const int32_t rightFlIdx = (int32_t)floor(rightIdx);
+        const int32_t rightClIdx = (int32_t)ceil(rightIdx);
+        const int32_t leftFlIdx = (int32_t)floor(leftIdx);
+        const int32_t leftClIdx = (int32_t)ceil(leftIdx);
+
+        assert(linearSize > 1);

```

Comment:
may wanna also assert that `radius>numeric_limits::min` or something

Reply:
you want to specifically assert that `raidus` is bigger than a denormalized float (hence the `bit_cast<float32_t>(numeric_limits<float32_t>::min)`

denormalized numbers can flush to 0, which means expression like `1+2*radius` can equal  0 which in turn will make your normalization factor inf